components/ai_chat/core/browser/utils.cc

```diff
@@ -253,4 +259,31 @@ SkBitmap ScaleDownBitmap(const SkBitmap& bitmap) {
   return scaled_bitmap;
 }
 
+GURL GetEndpointUrl(bool premium, const std::string& path) {
+  CHECK(!path.starts_with("/"));
+
+#if !defined(OFFICIAL_BUILD)
+  // If a runtime AI Chat URL is provided, use it.
+  std::string ai_chat_url =
+      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+          kAIChatServerUrl);
+  if (!ai_chat_url.empty()) {
+    GURL url = GURL(base::StrCat({ai_chat_url, "/", path}));
+    CHECK(url.is_valid()) << "Invalid API Url: " << url.spec();
+    return url;
+  }
+#endif
+
+  auto* prefix = premium ? "ai-chat-premium.bsg" : "ai-chat.bsg";
+  auto hostname = brave_domains::GetServicesDomain(
+      prefix, brave_domains::ServicesEnvironment::DEV);
```

Comment:
Is there a reason we want it to default to `DEV` here? Seems the parameter has a default value of PROD normally